mu-plugins/build-processes-demo-features/build-processes-demo-features.php

```diff
@@ -23,17 +25,24 @@
 
 defined( 'ABSPATH' ) || exit;
 
-// Load the features plugin's translated strings.
-function bpd_features_load_textdomain() {
-	load_muplugin_textdomain( 'build-processes-demo-features', dirname( plugin_basename( __FILE__ ) ) . '/languages' );
-}
-add_action( 'init', 'bpd_features_load_textdomain' );
+// Define plugin constants.
+define( 'BPD_FEATURES_VERSION', '0.1.0' );
+define( 'BPD_FEATURES_DIR', plugin_dir_path( __FILE__ ) );
+define( 'BPD_FEATURES_URL', plugin_dir_url( __FILE__ ) );
 
-// Load the features plugin's files.
-foreach ( glob( __DIR__ . '/includes/*.php' ) as $bpd_features_filename ) {
-	if ( preg_match( '#/includes/_#i', $bpd_features_filename ) ) {
-		continue; // Ignore files prefixed with an underscore.
-	}
+// Define minimum environment requirements.
+define( 'BPD_FEATURES_MIN_PHP', '8.1' );
+define( 'BPD_FEATURES_MIN_WP', '6.1' );
 
-	include $bpd_features_filename;
+// Include the rest of the features plugin's files if system requirements check out.
```

Comment:
What happens if requirements aren't met? I notice we have the correct settings in the plugin header, e.g `requires php` and `requires at least`, that should prevent the plugin from activating if those checks aren't met. Does that make these checks redundant, or perhaps a secondary fallback safety check? https://developer.wordpress.org/plugins/plugin-basics/header-requirements/

Reply:
Since this is a mu-plugin, the headers don't matter 😢 Even if this was a regular plugin, the headers only prevent it from activating but don't prevent it from running if it's already active and the PHP version is lowered. So the check always makes sense, in my opinion.

> What happens if requirements aren't met?

Right now, nothing. If this was a commercial plugin, I would definitely add an admin notice about it. It being internal, I'm not sure it's worth doing that, but I'm happy to defer to your opinion.

Reply:
✅  Thanks for the clear explaination on the checks, and also agreed that a dashboard notice has little value to us.